File: 프로그래머스/최소직사각형.py
def solution(sizes):
    w_sizes = []
    h_sizes = []

    for size in sizes:
        if size[0] > size[1]:
            w_sizes.append(size[0])
            h_sizes.append(size[1])
        else:
            w_sizes.append(size[1])
            h_sizes.append(size[0])

    answer = max(w_sizes) * max(h_sizes)

    return answer

File: 프로그래머스/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int width = 0;
        int height = 0;
        //sizes[n][0] = w
        //sizes[n][1] = h
        
        for(int i = 0; i < sizes.length; i++){
            
            if(sizes[i][0] < sizes[i][1]){//가로 세로 길이 비교해서 큰값을 가로로, 작은값을 세로로 swap
                int temp = sizes[i][0];
                sizes[i][0] = sizes[i][1];
                sizes[i][1] = temp;
            }
            
            if(width < sizes[i][0]){//가로길이 갱신
                width = sizes[i][0];
            }
            
            if(height < sizes[i][1]){//세로길이 갱신
                height = sizes[i][1];
            }
        }
        answer = width * height;
        return answer;
    }
}